File: poo/python/Heranca/ContaPoupanca.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 06/04/14
"""
from __future__ import unicode_literals
from __future__ import print_function

from Conta import Conta

class ContaPoupanca(Conta):

    # Construtor #
    def __init__(self, num):
        super(ContaPoupanca, self).__init__(num)

    # Redefinida #
    def atualiza(self, taxa):
        self._saldo += 3*(self._saldo * taxa)

File: poo/java/Heranca/ContaPoupanca.java
/*
 *  Programa escrito com propósito didático apenas.
 *  @author Nathan Cruz <nathan.cruz@unifesp.br>
 *  @created 04/04/14
 *  R.A.: 86492
 */

public class ContaPoupanca extends Conta{

    /* Construtor */
    public ContaPoupanca(int i){
        super(i);
    }

    @Override
    public void atualiza(double taxa){
        this.saldo += 3*(this.saldo * taxa);
    }
}
